test/batchReceiver.spec.ts

```diff
@@ -421,4 +423,533 @@ describe("ReceiveBatch from Queue/Subscription", function(): void {
+    }
+    should.equal(deferredMsgs.body, testMessages[0].body);
+    should.equal(deferredMsgs.messageId, testMessages[0].messageId);
+
+    await deferredMsgs.abandon();
+
+    await testPeekMsgsLength(subscriptionClient, 1);
+
+    if (!sequenceNumber) {
+      throw "Sequence Number can not be null";
+    }
+    const abandonedMsgs = await subscriptionClient.receiveDeferredMessage(sequenceNumber);
+    if (!abandonedMsgs) {
+      throw "No message received for sequence number";
+    }
+    should.equal(abandonedMsgs.body, testMessages[0].body);
+    should.equal(abandonedMsgs.messageId, testMessages[0].messageId);
+
+    await abandonedMsgs.complete();
+
+    await testPeekMsgsLength(subscriptionClient, 0);
+  });
+
+  it("Dead letter a message received from recevieDefferedMessage using queue", async function(): Promise<
+    void
+  > {
+    await queueClient.send(testMessages[0]);
+    const sequenceNumber = await deferMsgAndGetSequenceNum(queueClient);
+
+    const deferredMsgs = await queueClient.receiveDeferredMessage(sequenceNumber);
+    if (!deferredMsgs) {
+      throw "No message received for sequence number";
+    }
+    should.equal(deferredMsgs.body, testMessages[0].body);
+    should.equal(deferredMsgs.messageId, testMessages[0].messageId);
+
+    await deferredMsgs.deadLetter();
+
+    await testPeekMsgsLength(queueClient, 0);
+
+    const deadLetterQueuePath = Namespace.getDeadLetterQueuePathForQueue(queueClient.name);
+    const deadletterQueueClient = namespace.createQueueClient(deadLetterQueuePath);
+    const deadLetterMsgs = await deadletterQueueClient.receiveBatch(1);
+
+    should.equal(deadLetterMsgs.length, 1);
+    should.equal(deadLetterMsgs[0].body, testMessages[0].body);
+    should.equal(deadLetterMsgs[0].messageId, testMessages[0].messageId);
+
+    await deadLetterMsgs[0].complete();
+
+    await testPeekMsgsLength(deadletterQueueClient, 0);
+  });
+
+  it("Dead letter a message received from recevieDefferedMessage using subscription", async function(): Promise<
+    void
+  > {
+    await topicClient.send(testMessages[0]);
+    const sequenceNumber = await deferMsgAndGetSequenceNum(subscriptionClient);
+
+    const deferredMsgs = await subscriptionClient.receiveDeferredMessage(sequenceNumber);
+    if (!deferredMsgs) {
+      throw "No message received for sequence number";
+    }
+    should.equal(deferredMsgs.body, testMessages[0].body);
+    should.equal(deferredMsgs.messageId, testMessages[0].messageId);
+
+    await deferredMsgs.deadLetter();
+
+    await testPeekMsgsLength(subscriptionClient, 0);
+
+    const deadLetterSubscriptionPath = Namespace.getDeadLetterSubcriptionPathForSubcription(
+      topicClient.name,
+      subscriptionClient.subscriptionName
+    );
+
+    const deadletterSubscriptionClient = namespace.createSubscriptionClient(
+      deadLetterSubscriptionPath ? deadLetterSubscriptionPath : "",
+      subscriptionClient.subscriptionName
+    );
```

Comment:
We seem to be repeating this a lot :) First calling `Namespace.getDeadLetterSubcriptionPathForSubcription` then `namespace.createSubscriptionClient`. Maybe we can have a helper method in the test file that combines both?

Reply:
I'll make 2 helper methods one for queue and one for subscription.